src/component/vocabulary/VocabularySummary.tsx

```diff
@@ -114,6 +118,10 @@ export class VocabularySummary extends EditableComponent<
     });
   };
 
+  public onDocumentSave = (document: Document) => {
```

Comment:
I would replace the parameter type with `DocumentData`.  Otherwise it is a bit unintuitive why we create a new `Document` from the provided one.